### PR TITLE
RS-15119: Set tick distance to monthly intervals

### PR DESCRIPTION
**Pull request by chschan**

### CODE REVIEW

File: R/chartutils.R [hunk 1]
```diff
@@ -653,15 +653,27 @@ setDateTickDistance <- function(date.labels, num.maxticks)
...
-    else # whether to show month/day
+    else if (tmp.dist <= 0.9 * 31536000) { # on the scale of a year
+        # Check for monthly intervals which accounts for different number of days per month
+        nmonth <- round(tmp.dist/(86400*30))
+        if (!is.null(num.maxticks) && n > num.maxticks) 
+            nmonth <- (floor(n/num.maxticks) + 1) * nmonth
+        day.of.month <- as.numeric(format(date.labels, "%d"))
+        if (length(unique(day.of.month)) == 1)
+            return(paste0("M", nmonth))
+        offset <- 31 - as.numeric(min(day.of.month))
```
Comment (JustinCCYap) [inline R/chartutils.R:670]: Could you explain how this offset works? It seems to be an integer scalar from 0 to 3? How does adding it to `data..labels` make the day on month be unique in the example case?

Comment (chschan) [reply R/chartutils.R:670]: Its checking if its the same day counting backwards from the end of the month. In the example in the test, the dates are the last day of the month, so when we increment by one day, all the dates become "1".  
I'll add a comment saying "Checking dates relative to the end of the month" ?

Comment (JustinCCYap) [reply R/chartutils.R:670]: If `day.of.month` contains only 31, then the offset will be 0. How will the day be the 1st in the check below?

Comment (chschan) [reply R/chartutils.R:670]: As long as the offset is big enough to get you past the end of the month its ok - for the current example, setting the offset to 2 will still make all dates be on the 2nd of the next month

Comment (JustinCCYap) [reply R/chartutils.R:670]: OK I think I get it now

**Review by JustinCCYap — APPROVED** [review]

**Pull request merged**